docs/developer/reproducibility.md

```diff
@@ -0,0 +1,125 @@
+
+## Container image
+
+### Current limitations
+
+Our container image is currently not reproducible for the following main
+reasons:
+
+* We build PyMuPDF from source, since it's not available in Alpine Linux. The
+  result of this build is not reproducible. Note that PyMuPDF wheels are
+  available from PyPI, but there are no ARM wheels for the musl libc platforms.
+* Alpine Linux does not have a way to pin packages and their dependencies, and
+  does not retain old packages. There's a
+  [workaround](https://github.com/reproducible-containers/repro-pkg-cache)
+  to download the required packages and store them elsewhere, but then the
+  cached package downloads cannot be easily audited.
+
+## Proposed implementation
+
+We can take advantage of the
+[Debian snapshot archives](https://snapshot.debian.org/)
+and pin our packages by specifying a date. There's already
+[prior art](https://github.com/reproducible-containers/repro-sources-list.sh/)
+for that, thanks to the incredible work of @AkihiroSuda on
+[reproducible containers](https://github.com/reproducible-containers).
+As for PyMuPDF, it is available from the Debian repos, so we won't have to build
+it from source.
+
+Here are a few other obstacles that we need to overcome:
+* We currently download the
+  [latest gVisor version](https://gvisor.dev/docs/user_guide/install/#latest-release)
+  from a GCS bucket. Now that we have switched to Debian, we can take advantage
+  of their
+  [timestamped APT repos](https://gvisor.dev/docs/user_guide/install/#specific-release)
+  and download specific releases from those. An extra benefit is that such
+  releases are signed with their APT key.
+* We can no longer update the packages in the container image by rebuilding it.
+  We have to bump the dates in the Dockerfile first, which is a minor hassle,
+  but much more declarative.
+* The `repro-source-list-.sh` script uses the release date of the container
+  image. However, the Debian image is not updated daily (see
+  [newest tags](https://hub.docker.com/_/debian/tags)
+  in DockerHub). So, if we want to ship an emergency release, we have to
+  circumvent this limitation. A simple way is to trick the script by bumping the
+  date of the `/etc/apt/sources.list.d/debian.sources` and
+  `/etc/apt/sources.list` files.
+* While we talk about image reproducibility, we can't actually achieve the exact
+  same SHA-256 hash for two different image builds. That's because the file
+  timestamps in the image layers will differ, depending on when the build took
+  place. The rest of the image though (file contents, permissions, manifest)
+  should be byte-for-byte the same. A simple way to check this is with the
+  [`diffoci`](https://github.com/reproducible-containers/diffoci) tool, and
+  specifically this invocation:
+
+  ```
+  ./diffoci diff podman://<new_image_tag> podman://<old_image_tag> \
+      --ignore-timestamps --ignore-image-name --verbose
+  ```
+
+### Updating the image
```

Comment:
I feel we're missing a howto check reproducibility for old images. Maybe rephrasing a bit this section could help in this regard (as I believe it would be updating the version numbers / dates and rebuilding the image, then diffing)

Reply:
I already have a "Reproducing the image" section in the design doc, not sure if you've seen it. If it's not clear what the user should do, here's an update in the wording: aa710e8

Reply:
Yep, it's better as it explains how to do it for older images, thanks!